src/util/Collection.js

```diff
@@ -333,15 +333,32 @@ class Collection extends Map {
    * @returns {Collection}
    * @example
    * collection
-   *  .tap(user => console.log(user.username))
+   *  .each(user => console.log(user.username))
    *  .filter(user => user.bot)
-   *  .tap(user => console.log(user.username));
+   *  .each(user => console.log(user.username));
    */
-  tap(fn, thisArg) {
+  each(fn, thisArg) {
     this.forEach(fn, thisArg);
     return this;
   }
 
+  /**
+   * Runs a function on the collection and returns the collection.
+   * @param {Function} fn Function to execute
+   * @param {*} [thisArg] Value to use as `this` when executing function
+   * @returns {Collection}
+   * @example
+   * collection
+   *  .tap(coll => coll.size)
+   *  .filter(user => user.bot)
+   *  .tap(coll => coll.size)
+   */
+  tap(fn, thisArg) {
+    if (typeof thisArg !== 'undefined') fn = fn.bind(thisArg);
```

Comment:
Since we're calling the function immediately and not storing it for later, I'd say that using `fn.apply` or `fn.call` is the way to go, rather than `fn.bind`. Bind creates a new function object and in this case that's unnecessary.